### PR TITLE
common: split `osh/common/csmock_parser.py` 

### PR DESCRIPTION
**Pull request by lzaoral**
Resolves: https://gitlab.cee.redhat.com/covscan/covscan/-/issues/268

### CODE REVIEW

**Comment by siteshwar** [comment]
@lachmanfrantisek  @lbarcziova Testing farm check is broken on this pull request. You may want to debug it.

**Comment by lbarcziova** [comment]
hi @siteshwar ! Thanks for pointing this out, I will have a closer look at this tomorrow. Could this be only a flake (you can try to retry by commenting `/packit test`), or has this happened repeatedly/always?

**Comment by siteshwar** [comment]
> Could this be only a flake (you can try to retry by commenting /packit test), or has this happened repeatedly/always?

It's the only time I have seen this happening.

**Comment by lbarcziova** [comment]
I tracked it down and opened https://github.com/packit/packit-service/issues/2119 for this, thanks for letting us know!

**Comment by lzaoral** [comment]
Reassigning review to @rhyw because @kdudka is preoccupied with high priority issues.

File: osh/worker/csmock_runner.py [hunk 1]
```diff
@@ -1,32 +1,7 @@
...
-        "store-results-to": "/tmp/asd",
-        "time-created": "2014-09-01 18:30:19",
-        "time-finished": "2014-09-01 18:38:15",
-        "tool": "csmock",
-        "tool-args": "'/bin/csmock' '-t' 'cppcheck,gcc,clang' '--no-scan' '-r' 'fedora-21-x86_64' '-o' 'asd' '--force'",
-        "tool-version": "csmock-1.3.2.20140829.165742.ge16c941-1.fc21"
-    },
-    "defects": ""
-}
-"""
```
Comment (rhyw) [inline osh/worker/csmock_runner.py:26]: I think we still need somewhere to be able to know what's inside the json output, if inline doc is not preferred, an external link(i.e., point to some data declaration from the csmock repo) would be helpful too.

Unfortunately the repo doc doesn't tell: https://github.com/csutils/csmock/tree/main/doc

**Comment by rhyw** [comment]
I had a quick look, the patch literally move different classes to different modules, thus change should be promising. I also had a look at the spec file specifically the `/osh/common`, `worker/` declarations, all looks good to me.

Generally LGTM.

File: osh/worker/csmock_runner.py (hunk 1 above)
Comment (kdudka) [reply osh/worker/csmock_runner.py:26]: Yes, there is no such documentation in `csmock` yet.

Comment (lzaoral) [reply osh/worker/csmock_runner.py:26]: I'll update the docstrings with the current format of the JSON file with results.

Comment (lzaoral) [reply osh/worker/csmock_runner.py:26]: Addressed in  824beeba205847a66681f95503d7f457f77689de.

**Review by kdudka — APPROVED** [review]
Looks good.  Thanks!

**Pull request merged**